### ISSUE TITLE
Opt-out of installing dpl in deploy step

### ISSUE THREAD

**Issue by palewire** [travis-ci/travis-ci#7687]
I just the deploy step in my builds to publish static files with Python's boto library. I do not like dpl to do my deployment. But if RubyGems goes down, as documented in #7686, my builds crash.

I understand that RubyGems going down is not your responsibility, but it would be nice to be able to opt out of using it if I don't need it.

**Comment by BanzaiMan** [comment]
In that case, please use `after_success`.

**Comment by palewire** [comment]
@BanzaiMan, according to the documentation, ``after_success`` the build does not fail if the step crashes. That's something I consider essential. I also enjoy taking advantage of the branch conditions in ``deploy`` and would prefer to continue to do so.

**Comment by palewire** [comment]
Is there any reason that ``dpl`` must be included? I would wager that a singificant number of users aren't taking advantage of it.

**Comment by nipendar** [comment]
using after_success doesn't work, travis tries to install dpl gem and fails due to ssl_error

**Comment by BanzaiMan** [comment]
@nipendar If you drop `deploy`, it should not try installing `dpl`.

**Comment by BanzaiMan** [comment]
All deployment providers are currently implemented with `dpl`. Everyone relies on it.

**Comment by BanzaiMan** [comment]
You are right, though, to point out that we should be relying less on external resources when it comes to critical aspect of our service such as deployment.

I have had thoughts, but none concrete enough yet to carry out.

This incident certainly has raised the stakes in my mind.

**Comment by palewire** [comment]
@BanzaiMan Thanks for considering the idea. If you need any feedback or testing, please let me know. A paid travis-ci.com deployment routine is how we published all of our special news projects at [latimes.com/projects](http://latimes.com/projects/) and I'm eager to find ways to speed up the builds and make them more reliable.
